pkg/loop/internal/core/services/capability/capabilities.go

```diff
@@ -283,30 +283,36 @@ type triggerExecutableClient struct {
 }
 
 func (t *triggerExecutableClient) RegisterTrigger(ctx context.Context, req capabilities.TriggerRegistrationRequest) (<-chan capabilities.TriggerResponse, error) {
-	responseStream, err := t.grpc.RegisterTrigger(ctx, pb.TriggerRegistrationRequestToProto(req))
+	subCtx, cancel := context.WithCancel(context.WithoutCancel(ctx))
+
+	responseStream, err := t.grpc.RegisterTrigger(subCtx, pb.TriggerRegistrationRequestToProto(req))
```

Comment:
A helper can avoid all the repeated calls to `cancel()`:
```suggestion
	ch, err := t.registerTrigger(...)
	if err != nil {
		cancel()
		return nil, err
	}
```
```go
func (t *triggerExecutableClient) registerTrigger(...) (chan, error) {
	responseStream, err := t.grpc.RegisterTrigger(subCtx, pb.TriggerRegistrationRequestToProto(req))
	if err != nil {
		return nil, fmt.Errorf("error registering trigger: %w", err)
	}

	// In order to ensure the registration is successful, we need to wait for the first message from the server.
	// This will be an ack or error message. If the error is not nil, we return an error.
	ackMsg, err := responseStream.Recv()
	if err != nil {
		return nil, fmt.Errorf("failed to receive registering trigger ack message: %w", err)
	}

	if ackMsg.GetAck() == nil {
		return nil, errors.New(fmt.Sprintf("failed registering trigger: %s", ackMsg.GetResponse().GetError()))
	}

	ch, err := forwardTriggerResponsesToChannel(subCtx, t.Logger, req, responseStream.Recv)
	if err != nil {
		return nil, fmt.Errorf("failed to start forwarding messages from stream: %w", err)
	}
	return ch, nil
}
```